instance/import.go

```diff
@@ -23,27 +23,32 @@ func (s *Store) UpdateObservations(w http.ResponseWriter, r *http.Request) {
 	vars := mux.Vars(r)
 	instanceID := vars["instance_id"]
 	insert := vars["inserted_observations"]
+	eTag := getIfMatch(r)
 	logData := log.Data{"instance_id": instanceID, "inserted_observations": insert}
 
-	if err := func() error {
-		observations, err := strconv.ParseInt(insert, 10, 64)
-		if err != nil {
-			log.Event(ctx, "update imported observations: failed to parse inserted_observations string to int", log.ERROR, log.Error(err), logData)
-			return errs.ErrInsertedObservationsInvalidSyntax
-		}
+	observations, err := strconv.ParseInt(insert, 10, 64)
+	if err != nil {
+		log.Event(ctx, "update imported observations: failed to parse inserted_observations string to int", log.ERROR, log.Error(err), logData)
+		handleInstanceErr(ctx, errs.ErrInsertedObservationsInvalidSyntax, w, logData)
+		return
+	}
 
-		if err = s.UpdateObservationInserted(instanceID, observations); err != nil {
-			log.Event(ctx, "update imported observations: store.UpdateObservationInserted returned an error", log.ERROR, log.Error(err), logData)
-			return err
-		}
+	instance, err := s.GetInstance(instanceID, eTag)
```

Comment:
Does this need wrapping with :
AcquireInstanceLock()
etc as done in dimension/dimension.go ... about line 45'ish ?

Reply:
this is a similar situation as above (event.go) - I can add the lock, but the `UpdateObservationInserted` call will fail if ETag changed between GetInstance and UpdateObservationInserted

Reply:
Update: due to other calls requiring the lock, I will need to also acquire it here, even if this call itself wouldn't need it, to prevent concurrency issues with them.